plugins/builds.js

```diff
@@ -224,6 +224,56 @@ exports.plugin = {
                         })
                     }
                 }
+            },
+            {
+                method: 'DELETE',
+                path: '/builds/{id}/{artifact*}',
+                handler: async (request, h) => {
+                    const buildId = request.params.id;
+                    const { artifact } = request.params;
+                    const id = `${buildId}-${artifact}`;
+
+                    if (usingS3) {
+                        try {
+                            await awsClient.removeObject(id);
+                        } catch (err) {
+                            request.log([id, 'error'], `Failed to delete artifact: ${err}`);
+                            throw err;
+                        }
+                    } else {
+                        try {
+                            await cache.drop(id);
+                        } catch (err) {
+                            request.log([id, 'error'], `Failed to delete artifact: ${err}`);
+                            throw err;
+                        }
+                    }
+
+                    return h.response().code(204);
+                },
+                options: {
+                    description: 'Delete build artifacts zip',
```

Comment:
this endpoint can delete any artifacts.
```suggestion
                    description: 'Delete build artifacts',
```